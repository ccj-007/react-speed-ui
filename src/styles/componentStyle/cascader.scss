@import "../commonStyle/variables.scss";

.speed-cascader {
  position: relative;

  &-content {
    display: flex;
    position: absolute;
    bottom: -240%;
    left: 0;
    background: #fff;
    border: 1px solid #ddd;
  }


  &-column {
    width: 150px;
    display: flex;
    flex-direction: column;
  }


  &-item {
    height: 40px;
    display: flex;
    justify-content: space-around;
    align-items: center;
  }

  &-item:hover {
    cursor: pointer;
    color: $green;
    background: $gray-100;

    .fa-angle-right {
      color: $green;
    }
  }
}