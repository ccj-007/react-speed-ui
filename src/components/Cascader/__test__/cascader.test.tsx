import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Cascader } from '../../index';
import { options } from '../cascader.stories';

describe('test Cascader component', () => {
  it('should render', () => {
    const { asFragment } = render(
      <>
        <Cascader options={options} />
      </>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
