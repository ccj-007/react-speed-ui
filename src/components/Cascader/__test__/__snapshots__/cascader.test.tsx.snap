// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Cascader component should render 1`] = `
<DocumentFragment>
  <div
    class="speed-cascader"
  >
    <div
      class="speed-input-warp"
    >
      <input
        class="speed-input"
        data-testid="input"
        readonly=""
        style="padding-right: 30px;"
        value=""
      />
    </div>
  </div>
</DocumentFragment>
`;
